File: php/list_trunks.php
#!/usr/bin/env php
<?php
require '_client.php';
$q = [];
if (getenv('PAGE_SIZE')) {
    $q['page_size'] = getenv('PAGE_SIZE');
}
if (getenv('PAGE_OFFSET')) {
    $q['offset'] = getenv('PAGE_OFFSET');
}
if (getenv('TRUNK_SID')) {
    $q['trunk_sid'] = getenv('TRUNK_SID');
}
echo "Listing trunks...\n";
exo_get('/trunks', $q ?: null);

File: java/ListTrunks.java
public class ListTrunks {
    public static void main(String[] args) throws Exception {
        StringBuilder q = new StringBuilder();
        String ps = System.getenv("PAGE_SIZE");
        String off = System.getenv("PAGE_OFFSET");
        String ts = System.getenv("TRUNK_SID");
        if (ps != null && !ps.isEmpty()) q.append("page_size=").append(ps).append("&");
        if (off != null && !off.isEmpty()) q.append("offset=").append(off).append("&");
        if (ts != null && !ts.isEmpty()) q.append("trunk_sid=").append(ts).append("&");
        String qs = q.length() > 0 ? q.substring(0, q.length() - 1) : "";
        System.out.println("Listing trunks...");
        _Client.get("/trunks", qs);
    }
}
